graphix/gflow.py

```diff
@@ -461,3 +394,365 @@ def get_layers(l_k):
     for i in l_k.keys():
         layers[l_k[i]].append(i)
     return d, layers
+
+
+class GF2Solver:
+    """Solver for GF(2) linear equations
+
+    Generalization of well-known Gauss-Jordan elimination onto GF2 and non-square matrix.
+
+    Step 1. GF2Solver.forward_elimination
+        Conduct row elimination and column pivoting when necessary.
+        We get a standard form of n x m matrix.
+    Step 2. GF2Solver.backward_substitution
+        We get solutions for each node and kernel of the matrix.
```

Comment:
maybe add code example in the docstring (optional)?

Reply:
I'm afraid that a code example doesn't provide any more information than the current docstring because it's obvious.

Reply:
for example, something like this is useful
https://numpy.org/doc/stable/reference/generated/numpy.linalg.solve.html
this being obvious is quite subjective and it's always good to be overly clear how to use it.

Reply:
just use the example I had on my z3-based solver

Reply:
I initially implemented this class as an internal function of glow because this is quite a rare problem and contains a little complicated process, especially the part of `SolutionNode`. I wonder if it is worthwhile to explain the usage of this method in detail. What do you think about it?

Reply:
ok understood! I'll make this resolved.